lib/graphql/client/schema/object_type.rb

```diff
@@ -184,6 +190,12 @@ def initialize(data = {}, errors = Errors.new, definer = nil)
 
           @definer = definer
           @enforce_collocated_callers = source_definition && source_definition.client.enforce_collocated_callers
+
+          self.class::METHODS_TO_UNDEF.each do |method_name|
+            begin
+              singleton_class.undef_method(method_name)
+            rescue; end
```

Comment:
I tried checking with `single_class.method_defined?(method_name)` but while it was returning true for i.e `:method`, it was returning false for `:test`. I'm not sure what else to try here to avoid the exception handling.